libnetdata/queue/queue.c

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include "queue.h"
+
+/*
+ * @Input:
+ *      q / The queue instance that the item will be pushed into
+ * 	    i / The item which will be pushed into the queue
+ * @Return
+ *      If the node is pushed the queue, it returns with 1, otherwise 0 
+ */
+int queue_push(queue_t q, void* i)	
+{		
+	node temp;
+	temp = (node) malloc(sizeof(struct node_s));
+	if (temp == NULL){
+		return 0;
+	}
+	pthread_mutex_lock(&q->lock);
+	while(q->count == q->max){
+		if(q->blocked == 0){
+			free(temp);
+			pthread_mutex_unlock(&q->lock);
+			return 0;
+		}
+		pthread_cond_wait(&q->condf, &q->lock);
+	}
+	temp->item = i;
+	temp->next = NULL;
+	if(q->count != 0)		
+	{		
+		q->rear->next = (node)temp;
+	}
+	else{
+		q->front = temp;
+	}
+	q->rear = temp;
+	q->count++;
+
+	pthread_mutex_unlock(&q->lock);
+	pthread_cond_signal(&q->conde);
+	return 1;
+}
+
+/*
+ * @Input:
+ *      q / The queue instance that the item will be popped from
+ * @Return
+ *      Returns the node if it is popped succesfully, otherwise NULL
+ */
+void* queue_pop(queue_t q)			
+{			
+	void* temp = NULL;					
+	node temp2;	
+	pthread_mutex_lock(&q->lock);
+	while(q->count == 0)
+	{
+		if(q->blocked == 0){
+			pthread_mutex_unlock(&q->lock);
+			return temp;
+		}
+		pthread_cond_wait(&q->conde, &q->lock);
+	}
+	
+	temp = q->front->item;			
+	temp2 = q->front;				
+	q->front = q->front->next;		
+	free(temp2);					
+	if(q->front==NULL){
+		q->rear=NULL;
+	} 
+	q->count--;
+
+	pthread_mutex_unlock(&q->lock);
+	pthread_cond_signal(&q->condf);
+	return temp;
+}
+
+/*
+ * @Input:
+ *      max / The queue size.
+ * @Return
+ *      Returns the queue if it is created successfully, otherwise NULL
+ */
+queue_t queue_new(int max, int blocked){
+	queue_t q = NULL;
+	q = (queue_t) malloc(sizeof(struct queue_s));	
```

Comment:
You are not checking for out of memory
Use mallocz -- has build-in check for out of memory